transaction/Transaction.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_TRANSACTION_TRANSACTION_HPP_
+#define QUICKSTEP_TRANSACTION_TRANSACTION_HPP_
+
+#include <cstdint>
+
+namespace quickstep {
+
+namespace transaction {
+
+/** \addtogroup Transaction
+ *  @{
+ */
+
+/**
+ * @brief Type that represents id of a transaction.
+ **/
+using TransactionId = std::uint64_t;
```

Comment:
Let's use typedef instead of ``TransactionId``. Also, the general format of data types is ``transaction_id`` and not ``TransactionId``